include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp

```diff
@@ -0,0 +1,113 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_RootSolver__
+#define __OpenSpaceToolkit_Astrodynamics_RootSolver__
+
+#include <boost/math/tools/roots.hpp>
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Size.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+
+using ostk::core::types::Size;
+using ostk::core::types::Real;
+
+/// @brief                      A Root Solver is a tool for finding the roots of equations.
+class RootSolver
+{
+   public:
+    struct Solution
```

Comment:
Nice, similarly, I think we can have a RootSolver Settings (or Config) Struct (not necessarily now but later if you had more methods) to store all the settings and parameters for the rootSolvers. They usually have a bunch in common (maxIters, tols, etc)

Reply:
Yeah for sure. I'm not sure how this class will grow so we can definitely do that in the future.